### PR TITLE
python39Packages.nest-asyncio: 1.5.4 -> 1.5.5

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
Update to latest upstream release 1.5.5

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 170517` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.napari</li>
    <li>python310Packages.skein</li>
    <li>python39Packages.skein</li>
  </ul>
</details>
<details>
  <summary>19 packages failed to build:</summary>
  <ul>
    <li>python310Packages.atomman</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.pydeck</li>
    <li>python310Packages.spyder-kernels</li>
    <li>python39Packages.atomman</li>
    <li>python39Packages.augmax</li>
    <li>python39Packages.einops</li>
    <li>python39Packages.elegy</li>
    <li>python39Packages.modeled</li>
    <li>python39Packages.phik</li>
    <li>python39Packages.pydeck</li>
    <li>python39Packages.robotframework-tools</li>
    <li>python39Packages.spyder-kernels</li>
    <li>python39Packages.treex</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>streamlit</li>
    <li>wolfram-notebook</li>
  </ul>
</details>
<details>
  <summary>158 packages built:</summary>
  <ul>
    <li>coconut (python39Packages.coconut)</li>
    <li>ihaskell</li>
    <li>inkcut</li>
    <li>inkscape-extensions.inkcut</li>
    <li>jupyter</li>
    <li>napari (python39Packages.napari)</li>
    <li>opsdroid</li>
    <li>python310Packages.Nikola</li>
    <li>python310Packages.Pweave</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.bash_kernel</li>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.cassandra-driver</li>
    <li>python310Packages.coconut</li>
    <li>python310Packages.cufflinks</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.folium</li>
    <li>python310Packages.gremlinpython</li>
    <li>python310Packages.ilua</li>
    <li>python310Packages.ipydatawidgets</li>
    <li>python310Packages.ipykernel</li>
    <li>python310Packages.ipympl</li>
    <li>python310Packages.ipyparallel</li>
    <li>python310Packages.ipyvue</li>
    <li>python310Packages.ipyvuetify</li>
    <li>python310Packages.ipywidgets</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter-c-kernel</li>
    <li>python310Packages.jupyter-client</li>
    <li>python310Packages.jupyter-lsp</li>
    <li>python310Packages.jupyter-server-mathjax</li>
    <li>python310Packages.jupyter-sphinx</li>
    <li>python310Packages.jupyter_console</li>
    <li>python310Packages.jupyter_server</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterlab</li>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-lsp</li>
    <li>python310Packages.jupyterlab_launcher</li>
    <li>python310Packages.jupyterlab_server</li>
    <li>python310Packages.jupytext</li>
    <li>python310Packages.kmapper</li>
    <li>python310Packages.lightparam</li>
    <li>python310Packages.livelossplot</li>
    <li>python310Packages.metakernel</li>
    <li>python310Packages.napari-console</li>
    <li>python310Packages.nbclassic</li>
    <li>python310Packages.nbclient</li>
    <li>python310Packages.nbconflux</li>
    <li>python310Packages.nbconvert</li>
    <li>python310Packages.nbdime</li>
    <li>python310Packages.nbsmoke</li>
    <li>python310Packages.nbsphinx</li>
    <li>python310Packages.nbval</li>
    <li>python310Packages.nest-asyncio</li>
    <li>python310Packages.nix-kernel</li>
    <li>python310Packages.notebook</li>
    <li>python310Packages.notedown</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.onnx</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.papermill</li>
    <li>python310Packages.potentials</li>
    <li>python310Packages.pytorch-pfn-extras</li>
    <li>python310Packages.qiskit-aer</li>
    <li>python310Packages.qiskit-finance</li>
    <li>python310Packages.qiskit-ibmq-provider</li>
    <li>python310Packages.qiskit-ignis</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.qiskit-optimization</li>
    <li>python310Packages.qiskit-terra</li>
    <li>python310Packages.qtconsole</li>
    <li>python310Packages.skytemple-ssb-debugger</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.widgetsnbextension</li>
    <li>python39Packages.Nikola</li>
    <li>python39Packages.Pweave</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.bash_kernel</li>
    <li>python39Packages.batchspawner</li>
    <li>python39Packages.cassandra-driver</li>
    <li>python39Packages.cufflinks</li>
    <li>python39Packages.datashader</li>
    <li>python39Packages.dockerspawner</li>
    <li>python39Packages.ffcv</li>
    <li>python39Packages.folium</li>
    <li>python39Packages.gremlinpython</li>
    <li>python39Packages.ilua</li>
    <li>python39Packages.ipydatawidgets</li>
    <li>python39Packages.ipykernel</li>
    <li>python39Packages.ipympl</li>
    <li>python39Packages.ipyparallel</li>
    <li>python39Packages.ipyvue</li>
    <li>python39Packages.ipyvuetify</li>
    <li>python39Packages.ipywidgets</li>
    <li>python39Packages.jupyter</li>
    <li>python39Packages.jupyter-c-kernel</li>
    <li>python39Packages.jupyter-client</li>
    <li>python39Packages.jupyter-lsp</li>
    <li>python39Packages.jupyter-server-mathjax</li>
    <li>python39Packages.jupyter-sphinx</li>
    <li>python39Packages.jupyter_console</li>
    <li>python39Packages.jupyter_server</li>
    <li>python39Packages.jupyterhub</li>
    <li>python39Packages.jupyterhub-ldapauthenticator</li>
    <li>python39Packages.jupyterhub-systemdspawner</li>
    <li>python39Packages.jupyterhub-tmpauthenticator</li>
    <li>python39Packages.jupyterlab</li>
    <li>python39Packages.jupyterlab-git</li>
    <li>python39Packages.jupyterlab-lsp</li>
    <li>python39Packages.jupyterlab_launcher</li>
    <li>python39Packages.jupyterlab_server</li>
    <li>python39Packages.jupytext</li>
    <li>python39Packages.kmapper</li>
    <li>python39Packages.lightparam</li>
    <li>python39Packages.livelossplot</li>
    <li>python39Packages.metakernel</li>
    <li>python39Packages.moretools</li>
    <li>python39Packages.napari-console</li>
    <li>python39Packages.nbclassic</li>
    <li>python39Packages.nbclient</li>
    <li>python39Packages.nbconflux</li>
    <li>python39Packages.nbconvert</li>
    <li>python39Packages.nbdime</li>
    <li>python39Packages.nbsmoke</li>
    <li>python39Packages.nbsphinx</li>
    <li>python39Packages.nbval</li>
    <li>python39Packages.nest-asyncio</li>
    <li>python39Packages.nix-kernel</li>
    <li>python39Packages.notebook</li>
    <li>python39Packages.notedown</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.onnx</li>
    <li>python39Packages.osmnx</li>
    <li>python39Packages.papermill</li>
    <li>python39Packages.potentials</li>
    <li>python39Packages.pygmo</li>
    <li>python39Packages.pytorch-pfn-extras</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aer</li>
    <li>python39Packages.qiskit-finance</li>
    <li>python39Packages.qiskit-ibmq-provider</li>
    <li>python39Packages.qiskit-ignis</li>
    <li>python39Packages.qiskit-machine-learning</li>
    <li>python39Packages.qiskit-nature</li>
    <li>python39Packages.qiskit-optimization</li>
    <li>python39Packages.qiskit-terra</li>
    <li>python39Packages.qtconsole</li>
    <li>python39Packages.scikit-tda</li>
    <li>python39Packages.skytemple-ssb-debugger</li>
    <li>python39Packages.stytra</li>
    <li>python39Packages.vega</li>
    <li>python39Packages.widgetsnbextension</li>
    <li>python39Packages.zetup</li>
    <li>skytemple</li>
  </ul>
</details>

File: pkgs/development/python-modules/augmax/default.nix [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
-, lib
+, pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "augmax";
   version = "unstable-2022-02-19";
   format = "setuptools";
 
+  disabled = pythonOlder "3.7";
```
Comment (samuela) [inline pkgs/development/python-modules/augmax/default.nix:15]: why is this necessary? I don't think we have anything older than python3.7 packaged in nixpkgs?

File: pkgs/development/python-modules/augmax/default.nix [hunk 2]
```diff
@@ -19,13 +22,22 @@ buildPythonPackage rec {
     sha256 = "046n43v7161w7najzlbi0443q60436xv24nh1mv23yw6psqqhx5i";
   };
 
-  propagatedBuildInputs = [ einops jax ];
+  propagatedBuildInputs = [
+    einops
+    jax
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/augmax/default.nix:28]: no change here please leave as is

File: pkgs/development/python-modules/augmax/default.nix [hunk 3]
```diff
@@ -19,13 +22,22 @@ buildPythonPackage rec {
...
 
   # augmax does not have any tests at the time of writing (2022-02-19), but
   # jaxlib is necessary for the pythonImportsCheckPhase.
-  checkInputs = [ jaxlib ];
+  checkInputs = [
+    jaxlib
+  ];
```
Comment (samuela) [inline pkgs/development/python-modules/augmax/default.nix:34]: no change here please leave as is

File: pkgs/development/python-modules/augmax/default.nix [hunk 4]
```diff
@@ -1,16 +1,19 @@
-{ buildPythonPackage
+{ lib
+, buildPythonPackage
 , einops
 , fetchFromGitHub
 , jax
 , jaxlib
-, lib
+, pythonOlder
```
Comment (samuela) [inline pkgs/development/python-modules/augmax/default.nix:7]: please just leave order as is

**Review by samuela — CHANGES_REQUESTED** [review]
why do all these commits belong in one PR? they don't seem related

File: pkgs/development/python-modules/augmax/default.nix (hunk 2 above)
Comment (samuela) [reply pkgs/development/python-modules/augmax/default.nix:28]: mixing formatting and semantic changes in one commit pollutes the git blame

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 170517` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.napari</li>
    <li>python310Packages.optuna</li>
    <li>python310Packages.skein</li>
    <li>python39Packages.optuna</li>
    <li>python39Packages.skein</li>
  </ul>
</details>
<details>
  <summary>21 packages failed to build:</summary>
  <ul>
    <li>napari (python39Packages.napari)</li>
    <li>python310Packages.atomman</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.pydeck</li>
    <li>python310Packages.spyder-kernels</li>
    <li>python39Packages.atomman</li>
    <li>python39Packages.augmax</li>
    <li>python39Packages.einops</li>
    <li>python39Packages.elegy</li>
    <li>python39Packages.napari-console</li>
    <li>python39Packages.phik</li>
    <li>python39Packages.pydeck</li>
    <li>python39Packages.robotframework-tools</li>
    <li>python39Packages.spyder-kernels</li>
    <li>python39Packages.stytra</li>
    <li>python39Packages.treex</li>
    <li>sage</li>
    <li>sageWithDoc</li>
    <li>streamlit</li>
    <li>wolfram-notebook</li>
  </ul>
</details>
<details>
  <summary>158 packages built:</summary>
  <ul>
    <li>coconut (python39Packages.coconut)</li>
    <li>ihaskell</li>
    <li>inkcut</li>
    <li>inkscape-extensions.inkcut</li>
    <li>jupyter</li>
    <li>opsdroid</li>
    <li>python310Packages.Nikola</li>
    <li>python310Packages.Pweave</li>
    <li>python310Packages.ansible-kernel</li>
    <li>python310Packages.bash_kernel</li>
    <li>python310Packages.batchspawner</li>
    <li>python310Packages.cassandra-driver</li>
    <li>python310Packages.chainer</li>
    <li>python310Packages.coconut</li>
    <li>python310Packages.cufflinks</li>
    <li>python310Packages.dockerspawner</li>
    <li>python310Packages.folium</li>
    <li>python310Packages.gremlinpython</li>
    <li>python310Packages.ilua</li>
    <li>python310Packages.ipydatawidgets</li>
    <li>python310Packages.ipykernel</li>
    <li>python310Packages.ipympl</li>
    <li>python310Packages.ipyparallel</li>
    <li>python310Packages.ipyvue</li>
    <li>python310Packages.ipyvuetify</li>
    <li>python310Packages.ipywidgets</li>
    <li>python310Packages.jupyter</li>
    <li>python310Packages.jupyter-c-kernel</li>
    <li>python310Packages.jupyter-client</li>
    <li>python310Packages.jupyter-lsp</li>
    <li>python310Packages.jupyter-server-mathjax</li>
    <li>python310Packages.jupyter-sphinx</li>
    <li>python310Packages.jupyter_console</li>
    <li>python310Packages.jupyter_server</li>
    <li>python310Packages.jupyterhub</li>
    <li>python310Packages.jupyterhub-ldapauthenticator</li>
    <li>python310Packages.jupyterhub-systemdspawner</li>
    <li>python310Packages.jupyterhub-tmpauthenticator</li>
    <li>python310Packages.jupyterlab</li>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-lsp</li>
    <li>python310Packages.jupyterlab_launcher</li>
    <li>python310Packages.jupyterlab_server</li>
    <li>python310Packages.jupytext</li>
    <li>python310Packages.kmapper</li>
    <li>python310Packages.lightparam</li>
    <li>python310Packages.livelossplot</li>
    <li>python310Packages.metakernel</li>
    <li>python310Packages.napari-console</li>
    <li>python310Packages.nbclassic</li>
    <li>python310Packages.nbclient</li>
    <li>python310Packages.nbconflux</li>
    <li>python310Packages.nbconvert</li>
    <li>python310Packages.nbdime</li>
    <li>python310Packages.nbsmoke</li>
    <li>python310Packages.nbsphinx</li>
    <li>python310Packages.nbval</li>
    <li>python310Packages.nest-asyncio</li>
    <li>python310Packages.nix-kernel</li>
    <li>python310Packages.notebook</li>
    <li>python310Packages.notedown</li>
    <li>python310Packages.oauthenticator</li>
    <li>python310Packages.onnx</li>
    <li>python310Packages.osmnx</li>
    <li>python310Packages.papermill</li>
    <li>python310Packages.potentials</li>
    <li>python310Packages.pytorch-pfn-extras</li>
    <li>python310Packages.qiskit-aer</li>
    <li>python310Packages.qiskit-finance</li>
    <li>python310Packages.qiskit-ibmq-provider</li>
    <li>python310Packages.qiskit-ignis</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.qiskit-optimization</li>
    <li>python310Packages.qiskit-terra</li>
    <li>python310Packages.qtconsole</li>
    <li>python310Packages.skytemple-ssb-debugger</li>
    <li>python310Packages.vega</li>
    <li>python310Packages.widgetsnbextension</li>
    <li>python39Packages.Nikola</li>
    <li>python39Packages.Pweave</li>
    <li>python39Packages.ansible-kernel</li>
    <li>python39Packages.bash_kernel</li>
    <li>python39Packages.batchspawner</li>
    <li>python39Packages.cassandra-driver</li>
    <li>python39Packages.chainer</li>
    <li>python39Packages.cufflinks</li>
    <li>python39Packages.datashader</li>
    <li>python39Packages.dockerspawner</li>
    <li>python39Packages.ffcv</li>
    <li>python39Packages.folium</li>
    <li>python39Packages.gremlinpython</li>
    <li>python39Packages.ilua</li>
    <li>python39Packages.ipydatawidgets</li>
    <li>python39Packages.ipykernel</li>
    <li>python39Packages.ipympl</li>
    <li>python39Packages.ipyparallel</li>
    <li>python39Packages.ipyvue</li>
    <li>python39Packages.ipyvuetify</li>
    <li>python39Packages.ipywidgets</li>
    <li>python39Packages.jupyter</li>
    <li>python39Packages.jupyter-c-kernel</li>
    <li>python39Packages.jupyter-client</li>
    <li>python39Packages.jupyter-lsp</li>
    <li>python39Packages.jupyter-server-mathjax</li>
    <li>python39Packages.jupyter-sphinx</li>
    <li>python39Packages.jupyter_console</li>
    <li>python39Packages.jupyter_server</li>
    <li>python39Packages.jupyterhub</li>
    <li>python39Packages.jupyterhub-ldapauthenticator</li>
    <li>python39Packages.jupyterhub-systemdspawner</li>
    <li>python39Packages.jupyterhub-tmpauthenticator</li>
    <li>python39Packages.jupyterlab</li>
    <li>python39Packages.jupyterlab-git</li>
    <li>python39Packages.jupyterlab-lsp</li>
    <li>python39Packages.jupyterlab_launcher</li>
    <li>python39Packages.jupyterlab_server</li>
    <li>python39Packages.jupytext</li>
    <li>python39Packages.kmapper</li>
    <li>python39Packages.lightparam</li>
    <li>python39Packages.livelossplot</li>
    <li>python39Packages.metakernel</li>
    <li>python39Packages.modeled</li>
    <li>python39Packages.moretools</li>
    <li>python39Packages.nbclassic</li>
    <li>python39Packages.nbclient</li>
    <li>python39Packages.nbconflux</li>
    <li>python39Packages.nbconvert</li>
    <li>python39Packages.nbdime</li>
    <li>python39Packages.nbsmoke</li>
    <li>python39Packages.nbsphinx</li>
    <li>python39Packages.nbval</li>
    <li>python39Packages.nest-asyncio</li>
    <li>python39Packages.nix-kernel</li>
    <li>python39Packages.notebook</li>
    <li>python39Packages.notedown</li>
    <li>python39Packages.oauthenticator</li>
    <li>python39Packages.onnx</li>
    <li>python39Packages.osmnx</li>
    <li>python39Packages.papermill</li>
    <li>python39Packages.potentials</li>
    <li>python39Packages.pygmo</li>
    <li>python39Packages.pytorch-pfn-extras</li>
    <li>python39Packages.qiskit</li>
    <li>python39Packages.qiskit-aer</li>
    <li>python39Packages.qiskit-finance</li>
    <li>python39Packages.qiskit-ibmq-provider</li>
    <li>python39Packages.qiskit-ignis</li>
    <li>python39Packages.qiskit-machine-learning</li>
    <li>python39Packages.qiskit-nature</li>
    <li>python39Packages.qiskit-optimization</li>
    <li>python39Packages.qiskit-terra</li>
    <li>python39Packages.qtconsole</li>
    <li>python39Packages.scikit-tda</li>
    <li>python39Packages.skytemple-ssb-debugger</li>
    <li>python39Packages.vega</li>
    <li>python39Packages.widgetsnbextension</li>
    <li>python39Packages.zetup</li>
    <li>skytemple</li>
  </ul>
</details>

**Review by yl3dy — CHANGES_REQUESTED** [review]
@fabaff Could you please split this PR? E.g. the changes to `einops` aren't relevant to `nest-asyncio` and are mostly cosmetic (the issue with einops PyTorch tests occurs irregardless to `format = "setuptools"` and `disabled = pythonOlder "3.7"`).

**Comment by RaitoBezarius** [comment]
Already implemented upstream.
